services/sidecar/rollup/derivation/batch_builder.go

```diff
@@ -101,12 +102,15 @@ func (b *batchBuilder) Advance() {
 }
 
 // Tries to get the current batch.
-func (b *batchBuilder) getBatch(l1Head types.BlockID) ([]byte, error) {
+func (b *batchBuilder) getBatch(l1Head types.BlockID, currentLag uint64) ([]byte, error) {
+	// NOTE: 36 is # of l2 blocks within 6 l1 blocks/the confirmation period
+	lagTooGreat := currentLag + 36 >= b.cfg.GetMaxSafeLag()
```

Comment:
define param in config (or embed in the max safe lag directly for now)

also you can move this line to before L113